### PR TITLE
fix(ras-acc): use config flag to skip auth newsletters signup

### PR DESCRIPTION
**Pull request by chickenn00dle**
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

See https://github.com/Automattic/newspack-plugin/pull/3404 for details and testing steps!

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/modal-checkout/modal.js [hunk 1]
```diff
@@ -403,6 +373,36 @@ domReady( () => {
 							content = `<div class="order-details-summary ${ CLASS_PREFIX }__box ${ CLASS_PREFIX }__box--text-center"><p><strong>${ priceSummary }</strong></p></div>`;
 						}
 
+						// Set reader activation checkout data if available.
```
Comment (chickenn00dle) [inline src/modal-checkout/modal.js:376]: Moved the setting of reader activation data into the check for reader activation since it is only needed in these cases.

File: src/modal-checkout/modal.js [hunk 2]
```diff
@@ -403,6 +373,36 @@ domReady( () => {
...
+						const data = {};
```
Comment (chickenn00dle) [inline src/modal-checkout/modal.js:377]: https://github.com/Automattic/newspack-plugin/pull/3404 removes the need for the `is_pending_checkout` flag, so initializing empty here.

File: includes/class-modal-checkout.php [hunk 3]
```diff
@@ -444,7 +444,7 @@ public static function render_variation_selection() {
 		$products = array_keys( self::$products );
 		foreach ( $products as $product_id ) {
 			$product = wc_get_product( $product_id );
-			if ( ! $product->is_type( 'variable' ) ) {
+			if ( ! $product || ! $product->is_type( 'variable' ) ) {
```
Comment (chickenn00dle) [inline includes/class-modal-checkout.php:447]: Snuck this in to fix a random fatal I ran into when accessing a content restricted post with a donation block:

```
[09-Sep-2024 00:01:02 UTC] PHP Fatal error:  Uncaught Error: Call to a member function is_type() on false in /Users/raz/Sites/newspack/blocks/includes/class-modal-checkout.php:447
Stack trace:
#0 /Users/raz/Sites/newspack/sites/site/wp-includes/class-wp-hook.php(324): Newspack_Blocks\Modal_Checkout::render_variation_selection('')
#1 /Users/raz/Sites/newspack/sites/site/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#2 /Users/raz/Sites/newspack/sites/site/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 /Users/raz/Sites/newspack/sites/site/wp-includes/general-template.php(3081): do_action('wp_footer')
#4 /Users/raz/Sites/newspack/theme/newspack-theme/footer.php(74): wp_footer()
#5 /Users/raz/Sites/newspack/sites/site/wp-includes/template.php(810): require_once('/Users/raz/Site...')
#6 /Users/raz/Sites/newspack/sites/site/wp-includes/template.php(745): load_template('/Users/raz/Site...', true, Array)
#7 /Users/raz/Sites/newspack/sites/site/wp-includes/general-template.php(92): locate_template(Array, true, true, Array)
#8 /Users/raz/Sites/newspack/theme/newspack-theme/single-feature.php(68): get_footer()
#9 /Users/raz/Sites/newspack/sites/site/wp-includes/template-loader.php(106): include('/Users/raz/Site...')
#10 /Users/raz/Sites/newspack/sites/site/wp-blog-header.php(19): require_once('/Users/raz/Site...')
#11 /Users/raz/Sites/newspack/sites/site/index.php(17): require('/Users/raz/Site...')
#12 /Users/raz/.composer/vendor/laravel/valet/server.php(110): require('/Users/raz/Site...')
#13 {main}
  thrown in /Users/raz/Sites/newspack/blocks/includes/class-modal-checkout.php on line 447
```

Let me know if you prefer I move this into its own PR.

**Review by laurelfulford — APPROVED** [review]
:shipit: 

(I'm cool with the fatal fix here -- good catch!!)
